Bio-KBase-Auth/lib/Bio/KBase/AuthToken.pm

```diff
@@ -585,6 +592,54 @@ sub go_request {
 
 }
 
+sub _get_token {
+    my $self = shift @_;
+    my %p = @_;
+
+    my $json;
+    eval {
+	my $baseurl = $Bio::KBase::Auth::AuthSvcHost;
+#	my %headers;
+#        warn $p{'path'};
+	unless ($p{'path'}) {
+	    die "No path specified";
```

Comment:
This code is a huge mess.  For my first pass I didn't clean it up, but it's clear that it needs work to make more readable.  I'll get to that when I fix up how the config file gets read.